core/task/schedulerstate.go

```diff
@@ -208,10 +206,25 @@ func (state *schedulerState) Start(ctx context.Context) {
 		if state.err != nil {
 			err = state.err
 			log.WithField("error", err.Error()).Debug("scheduler quit with error, main state machine GO_ERROR")
-			state.sm.Event(context.Background(), "GO_ERROR", err) //TODO: use error information in GO_ERROR
+			state.sm.Event(context.Background(), "GO_ERROR", err) // TODO: use error information in GO_ERROR
 		} else {
 			log.Debug("scheduler quit, no errors")
 			state.sm.Event(context.Background(), "EXIT")
 		}
 	}()
 }
+
+func (state *schedulerState) CopyExecutor() *mesos.ExecutorInfo {
```

Comment:
can you rename to `CopyExecutorInfo` for clarity?

also, perhaps [`proto.Clone()`](https://pkg.go.dev/google.golang.org/protobuf/proto#Clone) might be faster than marshalling and unmarshalling?

Reply:
The reason why I did not use `Clone` originally was, that it did not build, hence the stupid marshalling/unmarshalling. I got error 
```
*mesos.ExecutorInfo does not implement protoreflect.ProtoMessage (missing method ProtoReflect)
but the object in defined in filename.pb.go
```
but now it works... ffs literally the same code I wonder what I did differently before (I even copied the same code as before :D)

Reply:
it could have been that you imported a wrong lib, I saw in GoLand that you could pick three different libraries with `proto.Clone`

Reply:
maybe... truth is, that I let lsp handle adding dependencies...